packages/kbn-expect/expect.js

```diff
@@ -647,6 +647,11 @@ function i (obj, showHidden, depth) {
       return stylize('null', 'null');
     }
 
+    // format sets like arrays
+    if (value instanceof Set) {
+      value = Array.from(value)
+    }
```

Comment:
The format doesn't know how to handle sets properly, so it applies the code for generic objects, which prints a failure message like this:
```
expected {} to sort of equal {}
```
The quickest way to get a proper output was treating the sets as arrays for formatting purposes, which prints out failure messages like this:
```
expected [ 'a', 'b', 'c' ] to sort of equal [ 'x', 'y', 'z' ]
```